Tools/build/compute-changes.py

```diff
@@ -0,0 +1,195 @@
+"""Determine which GitHub Actions workflows to run.
+
+Called by ``.github/workflows/reusable-change-detection.yml``.
+We only want to run tests on PRs when related files are changed,
+or when someone triggers a manual workflow run.
+This improves developer experience by not doing (slow)
+unnecessary work in GHA, and saves CI resources.
+"""
+
+from __future__ import annotations
+
+import os
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+
+TYPE_CHECKING = False
```

Comment:
Any reason you avoided importing from typing?

Reply:
Performance (typing is even slower in 3.14 with annotationlib)